src/filters/resize.js

```diff
@@ -14,25 +14,23 @@ module.exports = resize;
  */
 function resize(url, options, secret) {
 	return new Promise((resolve) => {
-		url = url.split('?');
+		url = decodeURIComponent(url);
+		url = url.replace(/(&|\?)accessToken=[a-f0-9]*/, '');
 
-		if (url.length <= 1) {
-			return resolve(url);
-		}
-
-		url = url[0];
 		options = Object.assign({}, options);
 
 		// __keywords is added by nunjucks
 		delete options.__keywords;
 
-		const resize = [];
+		// remove all instances of t[resize] parameters in URL before adding only new values
+		url = url.replace(/(&|\?)t\[resize\]\[\w*\]=(<|>|\^|!|%|@|\d)*/g, '');
```

Comment:
After some testing seems like the `url` module doesn't really work with our way of storing arrays in the query string. So regex might be the best solution then.